apps/meteor/app/livechat/server/hooks/markRoomRespondedAndSetVisitorAbandonment.ts

```diff
@@ -0,0 +1,62 @@
+import type { IOmnichannelRoom } from '@rocket.chat/core-typings';
+import { isOmnichannelRoom, isEditedMessage } from '@rocket.chat/core-typings';
+import { LivechatRooms } from '@rocket.chat/models';
+
+import { setPredictedVisitorAbandonmentTime } from '../../../../ee/app/livechat-enterprise/server/lib/Helper';
+import { callbacks } from '../../../../lib/callbacks';
+import { settings } from '../../../settings/server';
+
+callbacks.add(
+	'afterSaveMessage',
+	async (message, room) => {
+		if (!isOmnichannelRoom(room)) {
+			return message;
+		}
+
+		// skips this callback if the message was edited
+		if (!message || isEditedMessage(message)) {
+			return message;
+		}
+
+		// skips this callback if the message is a system message
+		if (message.t) {
+			return message;
+		}
+
+		// if the message has a token, it was sent by the visitor, so ignore it
+		if (message.token) {
+			return message;
+		}
+
+		// check if room is yet awaiting for response from visitor
+		if (!(typeof room.t !== 'undefined' && room.t === 'l' && room.waitingResponse)) {
```

Comment:
You can use the `isOmnichannelRoom` helper here

Reply:
In hindsight, this entire check is not required since we already have a check above verifying if this is an omnichannel room